collector/cpu_netbsd.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2015 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build !nocpu
+// +build !nocpu
+
+package collector
+
+import (
+	"bytes"
+	"errors"
+	"math"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+	"unsafe"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/sys/unix"
+
+	"howett.net/plist"
+)
+
+type clockinfo struct {
+	hz     int32 // clock frequency
+	tick   int32 // micro-seconds per hz tick
+	spare  int32
+	stathz int32 // statistics clock frequency
+	profhz int32 // profiling clock frequency
+}
+
+type cputime struct {
+	user float64
+	nice float64
+	sys  float64
+	intr float64
+	idle float64
+}
+
+func getCPUTemperatures() (map[int]float64, error) {
+	out, err := exec.Command("envstat", "-x").Output()
```

Comment:
We're not allowing to execute external commands in the node-exporter. So right right way would be to get the data from whereever envstat goes them.